lib/charms/operator_libs_linux/v0/sysctl.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handler for the sysctl config
+"""
+
+import logging
+import os
+import re
+import glob
+from subprocess import STDOUT, CalledProcessError, check_output
+from typing import Mapping, Dict
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "17a6cd4d80104d15b10f9c2420ab3266"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 2
+
+
+SYSCTL_DIRECTORY = "/etc/sysctl.d"
+SYSCTL_FILENAME = f"{SYSCTL_DIRECTORY}/95-juju-sysctl.conf"
+SYSCTL_HEADER = f"""# This config file was produced by sysctl lib v{LIBAPI}.{LIBPATCH}
+#
+# This file represents the output of the sysctl lib, which can combine multiple
+# configurations into a single file like.
+"""
```

Comment:
**todo** In the header of the `SYSCTL_FILENAME`, we should list the apps that are contributing to this file (as agreed).

Reply:
Added as part of the `_merge()` call.